src/PotentialEnergyEquationTerms.jl

```diff
@@ -186,4 +159,126 @@ end
 
 @inline g′z_ccc(i, j, k, grid, ρ, p) = (p.g / p.ρ₀) * ρ[i, j, k] * Zᶜᶜᶜ(i, j, k, grid)
 
+## Grid metrics from https://github.com/tomchor/Oceanostics.jl/issues/163#issuecomment-2012623824
+
+function MetricField(loc, grid, metric)
+    metric_operation = GridMetricOperation(loc, metric, grid)
+    metric_field = Field(metric_operation)
+    return compute!(metric_field)
+end
+
+VolumeField(grid, loc=(Center, Center, Center)) = MetricField(loc, grid, volume)
+  AreaField(grid, loc=(Center, Center, Nothing)) = MetricField(loc, grid, Az)
+
+"""
+    function OneDReferenceField(f::Field)
+Return a `OneDReferenceField` of the data in `f` and the corresponding `z✶` coordinate.
+The gridded data is first reshaped into a 1D `Array` then sorted. Returned is a new `Field`
+of this sorted data as well as the `z✶` `Field` that are both on a one-dimensional grid that
+has the same vertical extent as the original field (`f.grid.Lz`) but with the same _total_
+number of points of the original domain (`Nx * Ny * Nz`).
+The `z✶` `Field` is defined as
+```math
+\\frac{1}{A}\\int_{f\\mathrm{min}}^{f\\mathrm{max}} \\mathrm{d}V.
+```
+and is computed by cumulatively summing the 1D `Array` of grid volumes `ΔV`.
+**Note:** `OneDReferenceField` is currently only appropriate for grids that have uniform
+horizontal area.
+"""
+function OneDReferenceField(f::Field; rev = false)
+
+    area = sum(AreaField(f.grid))
+    volume_field = VolumeField(f.grid)
+    v = reshape(interior(volume_field), :)
+    field_data = reshape(interior(f), :)
+
+    p = sortperm(field_data; rev)
+    sorted_field_data = field_data[p]
+    z✶ = cumsum(v[p]) / area # divide by area at surface
+
+    grid_arch = f.grid.architecture
+    grid_size = prod(size(f.grid))
+    new_grid = RectilinearGrid(grid_arch, size = grid_size, z = (-f.grid.Lz, 0), topology=(Flat, Flat, Bounded))
+
+    sorted_field = CenterField(new_grid)
+    set!(sorted_field, reshape(sorted_field_data, size(new_grid)))
+    z✶_field = CenterField(new_grid)
+    set!(z✶_field, reshape(z✶, size(new_grid)))
+
+    return sorted_field, z✶_field
+
+end
+
+"""
+    $(SIGNATURES)
+
+Return a `kernelFunctionOperation` to compute an approximation to the `BackgroundPotentialEnergy`
```

Comment:
I think with the algorithm with discussed, this isn't an approximation anymore, correct?

Reply:
I thought the reshaping was an approximation but I think you might be right. Will have a double check now.

Reply:
It definitely was with the simple resorting we started with. But if you start with a distribution that's discrete, like we are since this is a numerical model on a grid, then I think reshaping doesn't introduce any further approximations, so I think we're okay removing the "approximation" from the dosctring. But I 100% certain. Just... 95% certain haha.

Reply:
Will do!